C/services/south/ingest.cpp

```diff
@@ -1082,11 +1083,20 @@ void Ingest::useFilteredData(OUTPUT_HANDLE *outHandle,
 		// down this will note cause a problem as all memory is recovered at process
 		// exit time.
 		ingest->m_data = new vector<Reading *>;
+		isBufferMemory = true;
 	}
 	ingest->m_data->insert(ingest->m_data->end(), newData->cbegin(), newData->cend());
 	
 	readingSet->clear();
 	delete readingSet;
+
+	// Recover buffer memory generated to ingest data
+	if (isBufferMemory)
```

Comment:
I do not believe this is safe as you have no idea what the ingest thread is doing with the memory, in particular Ingest::processQueue(). Once you have inserted the data into the ingest m_data queue that memory is owned by the ingest thread. It may work in some tests because of timing, but I don't think this is not a satisfactory solution.